### DISCUSSION TITLE
Difficulty making horizontal scroll line up

### DISCUSSION THREAD

**Discussion by HaveSpacesuit** [iTwin/iTwinUI-react#634] [Q&A]
I'm having trouble implementing iTwin table with many columns. The horizontal scroll will not line up unless I set the `min-width` to undefined or 0 on the table, and I add a specific width to each column. But adding a width to each column causes the table to not fill available space if using fewer columns.

**Column widths:** unset
**Table width:** set by iTwin
A few columns look nice
![image](https://user-images.githubusercontent.com/15036346/164513225-05296dfd-bb33-44e6-96cb-378db022a659.png)

But many columns don't line up, and the header doesn't scroll
![image](https://user-images.githubusercontent.com/15036346/164513583-bd9bd5e7-c95b-4377-be2e-21b3227efb52.png)

-----
**Column widths:** unset
**Table width:** manually styled as `unset !important`, overriding iTwin
Horizontal scroll works better, the header can scroll, but the columns are not lined up
![image](https://user-images.githubusercontent.com/15036346/164513867-9d4911b2-34b7-4b79-b69f-ec679e71136e.png)

-----
**Column widths:** each is set to 150
**Table width:** manually styled as `unset !important`
Everything scrolls and lines up nicely
![image](https://user-images.githubusercontent.com/15036346/164514101-19294beb-b260-42f7-96de-bf7dc9e7bfa3.png)

But with a few columns, the columns do not space out evenly, leaving the last column looking too large
![image](https://user-images.githubusercontent.com/15036346/164514242-6d0e902e-f372-47b3-ad70-792a2df23da6.png)

----
**Column widths:** each is set to 150
**Table width:** set by iTwin
Horizontal scroll does not work
![image](https://user-images.githubusercontent.com/15036346/164514427-0dbbfd66-e9e1-49fe-bc3b-7c64435622ee.png)

**Comment by mayank99** [comment]
Hi @HaveSpacesuit, what version of itwinui are you using? If you can post the result of `npm ls @itwin/itwinui-css` and `npm ls @itwin/itwinui-react`, that would be very helpful.

Initial horizontal scroll support was added in `@itwin/itwinui-react@1.28.0` and later fixed in `@itwin/itwinui-react@1.34.0` which uses `@itwin/itwinui-css@0.51.0`.

/cc @bentleyvk

**Reply by HaveSpacesuit** [reply]
I'm using 1.36.0. I did notice if I allow column resizing, as soon as I nudge any column everything snaps into place.

**Reply by HaveSpacesuit** [reply]
There is some problem with the text overflow. Removing white-space: nowrap htlps, except for the bit where I need the text to truncate. I'll keep playing at it.

**Comment by bentleyvk** [comment]
@HaveSpacesuit, can you prepare codesanbox with reproduction of this issue? Because I thought the problem might be too long strings but it is not causing this issue on my side.

**Reply by HaveSpacesuit** [reply]
@bentleyvk @mayank99 I've narrowed down the problem. If some columns have an undefined `width` and every column has a `minWidth`, the layout gets messed up. Here is a repo:

https://codesandbox.io/s/itwinui-react-minimal-example-forked-7in3qo

**Reply by HaveSpacesuit** [reply]
If you set all the widths to a number, the columns line up nicely. But when there aren't enough columns, the columns don't expand to fill available space.

![image](https://user-images.githubusercontent.com/15036346/165579330-3688cbc2-b5e4-4998-8a17-8a3ccb91deb2.png)

**Reply by bentleyvk** [reply]
Instead of `width`, I used `minWidth` and added sizes manually for nicer look. I also used `cellRenderer` for ellipsis. 
https://codesandbox.io/s/itwinui-react-minimal-example-forked-8f3ecq?file=/src/columns.ts
![image](https://user-images.githubusercontent.com/36186912/166212858-630304a0-4ccb-45c6-9c1d-6cd4ddca1411.png)
